### PR TITLE
Fix equations sign in Beam module

### LINKED ISSUE #20204: Beam module, wrong sign in two integration steps
I noticed that the first example at https://docs.sympy.org/latest/modules/physics/continuum_mechanics/beam_problems.html gives uncorrect results, but it seems there is a more general bug regarding **the sign of the integration of vertical loads**. As you can easily verify (see the code below), the shear **V** is calculated by integrating the vertical loads **q** without the needed sign change. In fact, regardless of user convention about coordinate system, the relation between V and q is **-q =dV/dx**, whereas the code seems to use **q=dV/dx**. This leads to wrong results, as in the above mentioned example and in the code below.

```python
from sympy import *
from sympy.physics.continuum_mechanics.beam import Beam
E,I = symbols('E,I')
b = Beam(1, E, I) # beam length=1
M,R = symbols('M,R')
b.apply_load(1,0,0) # uniform positive distributed vertical load=1
b.apply_load(M,0,-2) # add moment reaction at left end
b.apply_load(R,0,-1) # add force reaction at left end
b.bc_deflection.append((0,0)) # vertical displacement=0 constraint (cantilever beam fixed at left end)
b.bc_slope.append((0,0)) # rotation=0 constraint (cantilever beam fixed at left end)
b.solve_for_reaction_loads(M,R)
b.reaction_loads
b.plot_loading_results(subs={E: 20E9, I: 3.25E-6}) # the shear force diagram is wrong, it has positive slope instead of negative
```
Figure_1-opt

# Follow up

After more investigations I discovered that there is one more bug: in other integrations the equations **V = dM/dx** and **EJ v'' = M** are used, whereas in at least one of the two there should be a negative sign. On which of the two to put the negative sign depends on the coordinate system. I prepared a table showing that:

github-signes-table-opt

I think that the best way to quickly make the code working correctly is to choose the (**V = dM/dx**, **-EJ v'' = M**) option and explain in the docs that the user has NOT complete freedom in coordinate system choice.

In a quick and dirty fix attempt I made the following changes to `beam.py`:
- line 839 -->  return integrate(-self.load, x)
- line 1079 -->  slope_curve = integrate(-S.One/(E*I)*self.bending_moment(), x) + C3
- line 1177 -->  slope_curve = integrate(-self.bending_moment(), x) + C3

With the above corrections the first tests seem to give correct results, but before committing I guess someone more expert of the module should jump in the discussion.

Of course the documentation should be changed accordingly, to let the user know that only the last two coordinate systems (below in the table) are allowed.

To allow the user to choose also among the first 2 coordinate systems of the table above, an option should be added to the module, so the code would compute using (**-V = dM/dx**, **EJ v'' = M**) or (**V = dM/dx**, **-EJ v'' = M**) depending on which coordinate system the user wants to adopt.

### PR DESCRIPTION
**Pull request by nopria**
#### References to other Issues or PRs

Fixes #20204
See also #18045

#### Brief description of what is fixed or changed

Fix the sign of 2 equations used in the code:

1)  q =dV/dx   becomes   -q =dV/dx   (where q is distributed vertical load, V is shear, x is beam axis)
2)  EJ v'' = M   becomes   -EJ v'' = M   (where v is vertical displacement, M is moment, E is elastic modulus, J is moment of inertia)

The first equation is responsible (before fix) for wrong shear sign.
The second is responsible (before fix) for wrong results when used together with another equation present in code, **V = dM/dx**. In fact    **V = dM/dx**   and    **EJ v'' = M**   are incompatible, one of the two must have its sign changed, and changing sign to the latter equation allows to use the sign conventions most commonly used for internal forces ("discordant" shear and moment).

#### Other comments

- Optional TO DO: add an option to allow user to choose between "discordant" and "concordant" shear and moment convention.

#### Release Notes

* physics.continuum_mechanics
  * make beam module compute correctly internal forces

### CODE REVIEW

**Comment by nopria** [comment]
This commit fix a critical issue in Beam module, any idea on when it will be merged?

**Comment by nopria** [comment]
The horrible bug explained in issue #20204, which is solved by this pull request, still affects new Sympy version just published. It's really a pity that this commit has not been merged yet.

**Comment by oscarbenjamin** [comment]
I'm closing and reopening this to rerun the tests.

**Comment by oscarbenjamin** [comment]
@moorepants can you review this?

**Comment by moorepants** [comment]
@Psycho-Pirate looked at this in much more detail that I did. @Psycho-Pirate can you review?

**Comment by Psycho-Pirate** [comment]
> @Psycho-Pirate looked at this in much more detail that I did. @Psycho-Pirate can you review?

Sure, I will look into it.

File: sympy/physics/continuum_mechanics/beam.py
Comment (Psycho-Pirate) [inline sympy/physics/continuum_mechanics/beam.py:35]: I think this note might not be sufficient enough and beginner friendly in explaining what sign convention the user has to choose. Maybe adding an example might help. Something like this:
```
 For Example:   
    Given a right handed coordinate system with XYZ coordinates,
    the beam's length is assumed to be along the positive X axis.
    The loading forces are directed either in the positive or
    negative Y direction and the loading moments are about
    the positive or negative Z direction.
    The only thing that should be considered while choosing
    sign convention is that forces acting along positive
    Y direction(upward in this case) and moments acting along
    negative z direction(clockwise in this case) must have the same signs.
```
What are your thoughts?

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py:35]: I have seen a lot of new users make this kind of mistake and getting wrong results with the code.

**Review by Psycho-Pirate — CHANGES_REQUESTED** [review]
This looks great! I have added a few comments that must be addressed before merging.

File: sympy/physics/continuum_mechanics/beam.py
Comment (nopria) [reply sympy/physics/continuum_mechanics/beam.py:35]: Describing in a few words the allowed sign conventions it's not easy, maybe it's better to add an image like the following one. The curved arrow show the positive rotation. In orange is shown a possible (among many) alternate name for axes.

github-allowed-signes-convention-opt

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py:35]: Yes that would be better. Go ahead!

Comment (nopria) [reply sympy/physics/continuum_mechanics/beam.py]: I've updated the note about allowed sign conventions and added a description image, but I don't know how to check the result through Sphinx (I guess).

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: [See this.](https://docs.sympy.org/latest/documentation-style-guide.html?highlight=make%20html#build-the-documentation)
Please build the documentation and open the html file in a browser to see if the image is inserted or not. 
Also add a screenshot here of the webpage showing the image with the note.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst
Comment (Psycho-Pirate) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst:277]: Please fix this as well. It would be great if we could fix all the doc errors in one pull request.

Comment (nopria) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst:277]: Done, it was a typo in the problem description, now the description agrees with figure too.

File: sympy/physics/continuum_mechanics/beam.py
Comment (nopria) [reply sympy/physics/continuum_mechanics/beam.py]: Running `make html` I get an error:

```
rm -rf _build/logo
mkdir -p _build/logo
python ./generate_logos.py -d
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notail.svg
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notailtext.svg
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notext.svg
ERROR:root:Return code is not 0: Command: rsvg-convert C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\src/logo\sympy.svg -f png -o C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-160px.png -h 160 -w 160
ERROR:root:return code: 1
mingw32-make: *** [Makefile:144: logo] Error 1
```

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: There must be some unfulfilled dependencies. Let me build it on my system.

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: I tried building it. The image is present in the wrong directory. Please move it to `sympy/doc/src/modules/physics/continuum_mechanics/` and delete the one in the current directory.

File: sympy/physics/continuum_mechanics/beam.py
Comment (Psycho-Pirate) [inline sympy/physics/continuum_mechanics/beam.py:37]:
```suggestion

    .. image:: allowed-sign-conventions.png
```

File: sympy/physics/continuum_mechanics/beam.py
Comment (nopria) [reply sympy/physics/continuum_mechanics/beam.py]: I applied your commit suggestion and moved the image file to `sympy/doc/src/modules/physics/continuum_mechanics/`, but running `make html` I still get an error:

```
(nopria-sympy) C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc>make html
rm -rf _build/logo
mkdir -p _build/logo
python ./generate_logos.py -d
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notail.svg
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notailtext.svg
INFO:root: File saved: C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-notext.svg
ERROR:root:Return code is not 0: Command: rsvg-convert C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\src/logo\sympy.svg -f png -o C:\DATI\ingegneria\code\sympy-nopria\sympy-patch-1\doc\_build/logo\sympy-160px.png -h 160 -w 160
ERROR:root:return code: 1
mingw32-make: *** [Makefile:144: logo] Error 1
```

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: Screenshot from 2020-12-24 00-29-20
Its working now.

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: > I applied your [commit suggestion](https://github.com/sympy/sympy/pull/20431#discussion_r548012444) and moved the image file to `sympy/doc/src/modules/physics/continuum_mechanics/`, but running `make html` I still get an error:

I think you havent installed the dependencies required for sphinx properly. Have you installed librsvg?

**Review by Psycho-Pirate — APPROVED** [review]
Looks good to me! Great job!

File: sympy/physics/continuum_mechanics/beam.py
Comment (nopria) [reply sympy/physics/continuum_mechanics/beam.py]: No, I didn't. After installing librsvg I still get other errors, but I don't care as long as it worked for you. Glad to know `beam.py` will finally give correct results. Next I'd like to improve `apply_support` which currently isn't well designed. Have a look at issue #20404 if you have time.

Comment (Psycho-Pirate) [reply sympy/physics/continuum_mechanics/beam.py]: > No, I didn't. After installing librsvg I still get other errors, but I don't care as long as it worked for you. 

When i first built the documentation i had to install librsvg and imagemagick along with sphinx, as far as i remember. My terminal did tell me what dependencies i was missing. 
>Next I'd like to improve `apply_support` which currently isn't well designed. Have a look at issue #20404 if you have time.

Sure, I'll look into it!

**Comment by Psycho-Pirate** [comment]
@moorepants Please get this merged as i do not have write access.

**Comment by nopria** [comment]
> @moorepants Please get this merged as i do not have write access.

Any idea on why it hasn't been merged yet?

**Comment by moorepants** [comment]
@nopria, please be patient, we are all volunteers.

Thanks for this addition. Based on @Psycho-Pirate's review, merging.

**Pull request merged**
